src/promptflow-tools/tests/test_common.py

```diff
@@ -56,13 +56,23 @@ def test_parse_function_role_prompt(self):
         [
             ("system:\nthis is my function:\ndef hello", [
                 {'role': 'system', 'content': 'this is my function:\ndef hello'}]),
+            ("#system:\nthis is my ##function:\ndef hello", [
+                {'role': 'system', 'content': 'this is my ##function:\ndef hello'}]),
             (" \n system:\nthis is my function:\ndef hello", [
                 {'role': 'system', 'content': 'this is my function:\ndef hello'}]),
+            (" \n # system:\nthis is my function:\ndef hello", [
+                {'role': 'system', 'content': 'this is my function:\ndef hello'}]),
             ("user:\nhi\nassistant:\nanswer\nfunction:\nname:\nn\ncontent:\nc", [
                 {'role': 'user', 'content': 'hi'},
                 {'role': 'assistant', 'content': 'answer'},
                 {'role': 'function', 'name': 'n', 'content': 'c'}]),
+            ("#user :\nhi\n #assistant:\nanswer\n# function:\nname:\nn\n##content:\nc", [
```

Comment:
add a test case to for `##name`.

Reply:
update the existing test case to cover ##name